blog/k8s-probe.md

```diff
@@ -0,0 +1,308 @@
+
+Use `kubectl apply` to create the sample:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/skydive-project/skydive/v0.22.0/tests/k8s/pv-volume.yaml
+kubectl apply -f https://raw.githubusercontent.com/skydive-project/skydive/v0.22.0/tests/k8s/pv-claim.yaml
+kubectl apply -f https://raw.githubusercontent.com/skydive-project/skydive/v0.22.0/tests/k8s/pv-pod.yaml
+```
+When running the sample you will notice how the highlighted resources are RED
+(down) during creation phase, later turning WHITE (up) when operational. Final
+state depicted in the following figure:
+
+<p>
+  <a href="/assets/images/blog/k8s-storage.png" data-lightbox="storage" data-title="Skydive WebUI: Kubernetes Storage">
+    <img src="/assets/images/blog/k8s-storage.png" style="width:50%;"/>
+  </a>
+</p>
+
+## DevOps Automation
+
+Skydive has a REST API enabling automation which utilizes the Kubernetes
+topology model.
+
+We use wordpress sample and focus our view by:
+- Filter: `G.V().Has("K8s.Labels.app", "wordpress")`
+- Highlight: `G.V().Has("Type", "secret", "Name", "mysql-pass").In()`
+
+Let's create the wordpress sample (password **abc123**):
+
+```
+kubectl create secret generic mysql-pass --from-literal password=abc123
+kubectl apply -f https://raw.githubusercontent.com/skydive-project/skydive/v0.22.0/tests/k8s/mysql-deployment.yaml
+kubectl apply -f https://raw.githubusercontent.com/skydive-project/skydive/v0.22.0/tests/k8s/wordpress-deployment.yaml
+```
+
+Updating passwords on any landscape is standard practice and is typically
+governed by security regulations.
+
+Next we demonstrate updating the password by re-creating the secret resource
+(password **abc456**):
+
+```
+kubectl delete secret mysql-pass
+kubectl create secret generic mysql-pass --from-literal password=abc456
+```
+
+So we re-created `secret:mysql-pass` with **abc456** but the dependant pods
+were not re-created, and are still using the old password **abc123**.
+
+**Enter Skydive REST API.**
+
+Let's first install the Skydive binary:
+
+```
+wget -L0 https://github.com/skydive-project/skydive/releases/download/v0.22.0/skydive
+chmod +x skydive
+sudo mv skydive /usr/local/bin/skydive
+```
+
+And by utilizing the REST API we can delete all the outdated pods (password
+**abc123**) and let the deployment resources kick-in and create new replacement
+pods (password **abc456**):
+
+```
+skydive client query \
+	"G.V().Has('Name', 'mysql-pass').In().Has('Type', 'pod')" | \
+	jq '.[] | .Metadata.Name' | \
+	xargs -n 1 kubectl delete pod
+```
+
+The following figure captures the state of the system right after deleting the
+old pods (in WHITE before disappearing) by our script, and after the deployment
+resources automatically creates the new pods (in RED as they have not yet
+stabilized):
+
+<p>
+  <a href="/assets/images/blog/k8s-secret.png" data-lightbox="secret" data-title="Skydive WebUI: Kubernetes Secret">
+    <img src="/assets/images/blog/k8s-secret.png"/ style="width:50%;"/>
```

Comment:
The screenshot is pretty blurry. If you still have the original, could you please increase the quality ?

Reply:
it was captured from a video, in future I may re-create the screenshot in high resolution